browser/brave_wallet/brave_wallet_event_emitter_browsertest.cc

```diff
@@ -105,8 +96,57 @@ class BraveWalletEventEmitterTest : public InProcessBrowserTest {
     return std::move(json_rpc_service_);
   }
 
+  HostContentSettingsMap* host_content_settings_map() {
+    return HostContentSettingsMapFactory::GetForProfile(browser()->profile());
+  }
+
+  content::WebContents* web_contents() {
+    return browser()->tab_strip_model()->GetActiveWebContents();
+  }
+
+  url::Origin GetLastCommitedOrigin() {
+    return url::Origin::Create(web_contents()->GetLastCommittedURL());
+  }
+
+  void RestoreWallet() {
+    const char mnemonic[] =
+        "drip caution abandon festival order clown oven regular absorb "
+        "evidence crew where";
+    base::RunLoop run_loop;
+    keyring_service_->RestoreWallet(
+        mnemonic, "brave123", false,
+        base::BindLambdaForTesting([&](bool success) {
+          ASSERT_TRUE(success);
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+  }
+
+  void GetAddress(std::string* valid_address) {
+    ASSERT_NE(valid_address, nullptr);
+
+    base::RunLoop run_loop;
+    keyring_service_->GetKeyringInfo(
+        brave_wallet::mojom::kDefaultKeyringId,
+        base::BindLambdaForTesting(
+            [&](brave_wallet::mojom::KeyringInfoPtr keyring_info) {
+              *valid_address = "";
+              if (keyring_info->account_infos.size() > 0) {
+                *valid_address = keyring_info->account_infos[0]->address;
+              }
+              run_loop.Quit();
+            }));
+    run_loop.Run();
+  }
+
+  void SetSelectedAccount(const std::string& address) {
+    keyring_service_->SetSelectedAccount(
+        address, brave_wallet::mojom::CoinType::ETH, base::DoNothing());
```

Comment:
Don't use `DoNothing` here, even if we are not interested in the callback result, we still need use `BindLambdaForTesting` and `RunLoop` to wait for callback to be called.

Reply:
Why? `EvalJs` is blocking and wait for the event to happen

Reply:
```
SetSelectedAccount(address);

  auto result_first =
      EvalJs(contents, CheckForEventScript("received_account_changed_event"),
             content::EXECUTE_SCRIPT_USE_MANUAL_REPLY);
```

Reply:
actually it just waits for 100 milliseconds there, but anyways `SetSelectedAccount` is a sync call in keyring service

Reply:
As long as it is a async interface, we should treat it as async function in test. It is currently sync call underlying doesn't mean it won't change in the future.

Reply:
the only reason it's async is to pass back to js handler. It would be wrong if the internal call won't be sync. I do exactly an internal call. If it's ever changed, what I doubt, the test will fail and we will correct.

Reply:
I know for `SetSelectedAccount ` it would be no problem using it without waiting. My point is if this pattern of `base::DoNothing` get copied throughout all tests for other async API, those test cases would be flaky and hard to narrowing down. I've seen it happened so that is why I want to stop people creating test code like this.

Reply:
ok, I will do that, but it will be redundant in that case as you mentioned about to avoid redundancy in the above comment about `WaitForLoadStop`

Reply:
done!